### PR TITLE
Fix/attention mask mismatch

### PR DESCRIPTION
**Pull request by KatharinaJacoby**
The issue seems to come from the concatenation of the virtual attention mask, which changes the dimensions of the attention_mask and causes a mismatch with per_token_entropy.
Checking that the dimensions of per_token_entropy and attention_mask match before multiplication. I changed the code block so the attention_mask is truncated to match the sequence length of per_token_entropy. 
And added a unit test:
-set up common test data in the setUp method.
-test the scenario where the attention mask and per-token entropy have matching dimensions.
-test the scenario where the attention mask and per-token entropy have mismatched dimensions.
-test the scenario where position_ids is used instead of attention_mask.
-test the scenario where neither attention_mask nor position_ids is present, expecting a ValueError.

### CODE REVIEW

**Comment by pramodith** [comment]
Hey @KatharinaJacoby I think another PR that I opened last week resolves the issue that your PR covers too. #4196

**Comment by qgallouedec** [comment]
closing in favor of #4196, thanks a lot @KatharinaJacoby and @pramodith
